### DISCUSSION TITLE
Mismatch between Displayed Volume and Squeezeplay alsa device volume

### DISCUSSION THREAD

**Discussion by DanielHabenicht** [music-assistant/support#2746] [Q&A] [answered]
I hooked up a Pi with squeezelite to music assistant with a dummy alsa controller to control a hardware volume mixer. 
This led me to discover that the percentages shown in the UI do not match with the percentages of the alsa control. 
Here are some example values: 

0%:
```bash
Simple mixer control 'Master',0
  Capabilities: volume cswitch
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: -50 - 100
  Front Left: -50 [0%] [-45.00dB] Capture [off]
  Front Right: -50 [0%] [-45.00dB] Capture [off]
```

26%
```bash
Simple mixer control 'Master',0
  Capabilities: volume cswitch
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: -50 - 100
  Front Left: -22 [19%] [-36.60dB] Capture [off]
  Front Right: -22 [19%] [-36.60dB] Capture [off]
```

50%
```
Simple mixer control 'Master',0
  Capabilities: volume cswitch
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: -50 - 100
  Front Left: 18 [45%] [-24.60dB] Capture [off]
  Front Right: 18 [45%] [-24.60dB] Capture [off]
```

74%
```
Simple mixer control 'Master',0
  Capabilities: volume cswitch
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: -50 - 100
  Front Left: 58 [72%] [-12.60dB] Capture [off]
  Front Right: 58 [72%] [-12.60dB] Capture [off]
```

100%
```bash
Simple mixer control 'Master',0
  Capabilities: volume cswitch
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: -50 - 100
  Front Left: 100 [100%] [0.00dB] Capture [off]
  Front Right: 100 [100%] [0.00dB] Capture [off]
```

Commands used: 
```bash
sudo modprobe snd-dummy
squeezelite
amixer -c 0 get Master
```

The same happens if `-X` is used for squeezelite. Just that the values are skewed in another way: 
0% - 0
26% - 48
50% - 65
74% - 81
100% - 100

Is there some custom logic behind it? Can it be turned off?

**Comment by OzGav** [comment]
@marcelveldt ?

**Reply by marcelveldt** [reply]
I have no idea, slimproto volume control is one big black box

**Reply by DanielHabenicht** [reply]
Ahh, I think I can see why: https://github.com/home-assistant-libs/aioslimproto/blob/main/aioslimproto/volume.py

**Reply by DanielHabenicht** [reply]
@marcelveldt can I assume that the `volume_level` entering the blackbox here is between 0-100 in this function?
https://github.com/music-assistant/server/blob/0624393f0dc9ffcd8326c2d90994dd79af15b3ea/music_assistant/server/providers/slimproto/__init__.py#L525-L532

**Reply by marcelveldt** [reply, marked as answer]
Not really... its in the aioslimproto library
